pkg/cloudprovider/cache/cloudprovidercache.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type CloudproviderCache struct {
+	cache *gocache.Cache
+}
+
+// New returns a new cloudproviderCache
+func New() *CloudproviderCache {
+	return &CloudproviderCache{cache: gocache.New(gocache.NoExpiration, 5*time.Minute)}
```

Comment:
Are you sure you want cache keys to never expire?
This means any change on the cloud provider side which might lead to a failing validation will never be caught - leading to errors during the Create or funky behavior

Reply:
`5min`
That should be enough to cover the rate-limits we're facing

Reply:
Updated